Python/ceval.c

```diff
@@ -2121,9 +2121,13 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
             return 1;
         }
         Py_DECREF(w);
-        _PyErr_Format(tstate, PyExc_ValueError,
-                      "too many values to unpack (expected %d)",
-                      argcnt);
+        // TODO: we don't own v but we do own argcnt, that isn't properly handled yet
+        PyObject *args = Py_BuildValue("OO", v, PyLong_FromLong(argcnt));
```

Comment:
I think you can do something like that:

```C
PyObject *args = Py_BuildValue("Oi", v, argcnt);
if (args) {
	PyErr_SetObject(PyExc_UnpackError, args);
	Py_DECREF(args);
}
```

The `goto Error` would handle the case `args = NULL`.

Reply:
The `PyLong_FromLong` call can be moved into `UnpackError_init`, that's much better. Thanks.